crates/mpc-protocols/src/mpc_service.rs

```diff
@@ -0,0 +1,141 @@
+use crate::mpc_events::{CreatedProofMPCEvent, MPCEvent};
+use log::{debug, info};
+use pera_types::base_types::ObjectID;
+use pera_types::event::Event;
+use std::cell::RefCell;
+use std::collections::{HashMap, VecDeque};
+use tokio::sync::{mpsc, Mutex, RwLock};
+
+// TODO (#253): Make this configurable from a config file
+const MAX_ACTIVE_MPC_INSTANCES: usize = 100;
+
+pub enum MPCInput {
+    InitEvent(CreatedProofMPCEvent),
+    Message,
+}
+
+struct MPCInstance {
+    status: MPCSessionStatus,
+    /// The channel to send message to this instance
+    input_receiver: Option<mpsc::Sender<MPCInput>>,
```

Comment:
why is this an option?

Reply:
Because it can be none when the instance is pending. This entire channels logic has been removed in [258](https://github.com/dwallet-labs/dwallet-network/pull/258) anyway